hack/make/build-rpm

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+set -e
+
+DEST=$1
+
+# subshell so that we can export PATH without breaking other things
+(
+	source "$(dirname "$BASH_SOURCE")/.integration-daemon-start"
+
+	# TODO consider using frozen images for the dockercore/builder-rpm tags
+		
+	rpmVersion="${VERSION//-/_}"
+	# if we have a "-dev" suffix or have change in Git, let's make this package version more complex so it works better
+	if [[ "$VERSION" == *-dev ]] || [ -n "$(git status --porcelain)" ]; then
+		gitUnix="$(git log -1 --pretty='%at')"
+		gitDate="$(date --date "@$gitUnix" +'%Y%m%d.%H%M%S')"
+		gitCommit="$(git log -1 --pretty='%h')"
+		gitVersion="git${gitDate}.0.${gitCommit}"
+		# gitVersion is now something like 'git20150128.112847.0.17e840a'
+		rpmVersion="${rpmVersion}_${gitVersion}"
```

Comment:
so when i try to use - in Version: everything explodes :(

On Fri, May 1, 2015 at 2:42 PM, Tianon Gravi <notifications@github.com>
wrote:

> In hack/make/build-rpm
> <https://github.com/docker/docker/pull/12917#discussion_r29535906>:
>
> > +
> > +# subshell so that we can export PATH without breaking other things
> > +(
> > +	source "$(dirname "$BASH_SOURCE")/.integration-daemon-start"
> > +
> > +	# TODO consider using frozen images for the dockercore/builder-rpm tags
> > +		
> > +	rpmVersion="${VERSION//-/_}"
> > +	# if we have a "-dev" suffix or have change in Git, let's make this package version more complex so it works better
> > +	if [[ "$VERSION" == *-dev ]] || [ -n "$(git status --porcelain)" ]; then
> > +		gitUnix="$(git log -1 --pretty='%at')"
> > +		gitDate="$(date --date "@$gitUnix" +'%Y%m%d.%H%M%S')"
> > +		gitCommit="$(git log -1 --pretty='%h')"
> > +		gitVersion="git${gitDate}.0.${gitCommit}"
> > +		# gitVersion is now something like 'git20150128.112847.0.17e840a'
> > +		rpmVersion="${rpmVersion}_${gitVersion}"
>
> So, I went and read some of the Fedora packaging guides info and now I'm
> even more confused than I was before. They're sprinkled liberally with "-"
> in version numbers, contrary to random person on stackoverflow claiming
> that "-" needs to be "_". It appears that "%{checkout}" is what we should
> set, and it should be something like "YYYYMMDDgitSHORT-HASH" ?
> https://fedoraproject.org/wiki/Packaging:NamingGuidelines#Package_Versioning
> led me to:
> https://fedoraproject.org/wiki/Packaging:NamingGuidelines#NonNumericRelease
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/12917/files#r29535906>.
>